File: src/main/java/org/chousik/collection/validators/ValidatorCoordinatesX.java
package org.chousik.collection.validators;

import org.chousik.exception.InvalidDataError;

/**
 * Валидатор координаты X
 */
public class ValidatorCoordinatesX implements IValidator<Float> {

    /**
     * Проверяет число на соответствие диапазону
     *
     * @param aFloat число
     * @throws InvalidDataError     если число не соответствует диапазону
     * @throws NullPointerException если число null
     */
    @Override
    public void valide(Float aFloat) throws InvalidDataError, NullPointerException {
        if (aFloat == null) {
            throw new NullPointerException();
        }
        if (aFloat <= -645) {
            throw new InvalidDataError();
        }
    }
}

File: common/src/main/kotlin/collection/validators/ValidatorCoordinatesX.kt
package org.chousik.collection.validators

import exeption.InvalidDataError


class ValidatorCoordinatesX : IValidator<Float?> {
    private val const = -645
    override fun valid(t: Float?) {
        if (t == null){
            throw NullPointerException()
        }
        if (t <= const) {
            throw InvalidDataError()
        }
    }
}